mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -372,6 +375,60 @@ impl Drop for Nexus {
     }
 }
 
+extern "C" fn nvmf_create_snapshot_hdlr(req: *mut spdk_nvmf_request) -> i32 {
+    debug!("nvmf_create_snapshot_hdlr {:?}", req);
+
+    let subsys = unsafe { spdk_sys::spdk_nvmf_request_get_subsystem(req) };
+    if subsys.is_null() {
+        debug!("subsystem is null");
+        return -1;
+    }
+
+    /* Only process this request if it has exactly one namespace */
+    if unsafe { spdk_sys::spdk_nvmf_subsystem_get_max_nsid(subsys) } != 1 {
+        debug!("multiple namespaces");
+        return -1;
+    }
+
+    /* Forward to first namespace if it supports NVME admin commands */
+    let mut bdev: *mut spdk_bdev = std::ptr::null_mut();
+    let mut desc: *mut spdk_bdev_desc = std::ptr::null_mut();
+    let mut ch: *mut spdk_io_channel = std::ptr::null_mut();
+    let rc = unsafe {
+        spdk_sys::spdk_nvmf_request_get_bdev(
+            1, req, &mut bdev, &mut desc, &mut ch,
+        )
+    };
+    if rc != 0 {
+        /* No bdev found for this namespace. Continue. */
+        debug!("no bdev found");
+        return -1;
+    }
+
+    let bd = Bdev::from(bdev);
+    if let Some(replica) = Replica::from_bdev(&bd) {
+        let cmd = unsafe { &*spdk_sys::spdk_nvmf_request_get_cmd(req) };
+        let snapshot_time = unsafe {
+            cmd.__bindgen_anon_1.cdw10 as u64
+                | (cmd.__bindgen_anon_2.cdw11 as u64) << 32
+        };
+        let snapshot_name = format!("{}-snap-{}", bd.name(), snapshot_time);
+        replica.create_snapshot(req, &snapshot_name);
+        1 // SPDK_NVMF_REQUEST_EXEC_STATUS_ASYNCHRONOUS
+    } else {
+        -1
```

Comment:
I think its better to run errno variants here. for example if this is not a replica we could return Errono::ENOSUP as we need to have an lvol to be able to snap it to begin with.

Reply:
For this and the next comment, the caller nvmf_ctrlr_process_admin_cmd doesn't care about <0 return codes from custom admin command handlers, it simply falls through and ends up being handled as an unsupported admin opcode.